### PR TITLE
Works on two phase commit support

### PR DESCRIPTION
**Pull request by endurox-dev**
Hi, so there up to date patch for go-oci8 with two phase commit.

### CODE REVIEW

File: oci8.go [hunk 1]
```diff
@@ -399,7 +424,8 @@ type OCI8Conn struct {
 	transactionMode      C.ub4
 	inTransaction        bool
 	enableQMPlaceholders bool
-	closed               bool
+	is_xa            bool
+    closed               bool
```
Comment (mattn) [inline oci8.go]: please fix indentation

File: oci8.go [hunk 2]
```diff
@@ -661,43 +696,71 @@ func (d *OCI8Driver) Open(dsnString string) (connection driver.Conn, err error)
...
+        	(*C.OraText)(unsafe.Pointer(puser)),
+        	C.ub4(len(dsn.Username)),
+        	(*C.OraText)(unsafe.Pointer(ppass)),
+        	C.ub4(len(dsn.Password)),
+        	(*C.OraText)(unsafe.Pointer(phost)),
+        	C.ub4(len(dsn.Connect))); rv.rv != C.OCI_SUCCESS && rv.rv != C.OCI_SUCCESS_WITH_INFO {
+        	return nil, ociGetError(rv.rv, conn.err)
+        } else {
+        	conn.svc = rv.ptr
+        }
```
Comment (mattn) [inline oci8.go]: please fix indentation

File: oci8.go [hunk 3]
```diff
@@ -216,6 +216,30 @@ WrapOCIEnvCreate(ub4 mode, size_t extra) {
   return vvv;
 }
 
+//madars.vitolins@gmail.com XA support:
```
Comment (mattn) [inline oci8.go]: remove this line

File: oci8.go [hunk 4]
```diff
@@ -216,6 +216,30 @@ WrapOCIEnvCreate(ub4 mode, size_t extra) {
...
+  ret2ptr vvv = {NULL, NULL, 0};
+
+#ifdef OCI8_ENABLE_XA
+  vvv.ptr = (dvoid *)xaoEnv(NULL);
+#endif
+
+  return vvv;
+}
+
+//madars.vitolins@gmail.com XA support:
```
Comment (mattn) [inline oci8.go]: ditto

**Comment by endurox-dev** [comment]
All fixed, will create new pull request. 
"OCI8_ENABLE_XA" is a new C define which will enable XA OCI calls from go code. As you told before that there might be some issues in some environments, thus user have option to enable XA or disable.

For example:
`
#ifdef OCI8_ENABLE_XA
  vvv.ptr = (dvoid *)xaoEnv(NULL);
#endif
`

**Pull request closed without merging**